### PR TITLE
Feature/handle favorite

### PR DESCRIPTION
**Pull request by at-haivo**
add feature favorite video

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: Youtube/Controllers/FavoriteVC/FavoriteViewController.swift [hunk 1]
```diff
@@ -16,32 +16,70 @@ class FavoriteViewController: BaseViewController {
...
+        }
+    }
+
+    // MARK: - delete video from notification
+    @objc private func deleteFavoriteFromDetail(notification: NSNotification) {
+        if let videoWantdelete = notification.object as? Video {
+            ReadFromRealm.deleteVideo(videoWantdelete)
+            self.loadRealmData()
+        }
+    }
```
Comment (at-daiho) [inline Youtube/Controllers/FavoriteVC/FavoriteViewController.swift]: WARNING - There should be a space after //

File: Youtube/Controllers/FavoriteVC/FavoriteViewController.swift [hunk 2]
```diff
@@ -16,32 +16,90 @@ class FavoriteViewController: BaseViewController {
...
+                    self.loadRealmData()
+                    self.tableView.reloadData()
+                }
+            } catch {}
+        }
+    }
+
+    @objc private func deleteFavoriteFromDetail(notification: NSNotification) {
+        if let videoWantdelete = notification.object as? Video {
+            do {
```
Comment (at-daiho) [inline Youtube/Controllers/FavoriteVC/FavoriteViewController.swift]: WARNING - There should be a space after //

File: Youtube/Controllers/FavoriteVC/FavoriteViewController.swift [hunk 3]
```diff
@@ -73,12 +97,6 @@ extension FavoriteViewController: UITableViewDataSource {
         return true
     }
```
Comment (at-daiho) [inline Youtube/Controllers/FavoriteVC/FavoriteViewController.swift]: WARNING - There should be a space after //

File: Youtube/Controls/Cells/VideoTableViewCell.swift [hunk 4]
```diff
@@ -26,6 +26,12 @@ class VideoTableViewCell: UITableViewCell {
         if let URL = URL(string: video.thumbnail) {
             thumbnailImageView.hnk_setImageFromURL(URL)
         }
+        if video.isFavorite {
+            favoriteButton.setImage(#imageLiteral(resourceName: "ic_bt_video_like"), for: .normal)
+        } else {
+            favoriteButton.isHidden = true
```
Comment (at-haivo) [inline Youtube/Controls/Cells/VideoTableViewCell.swift]: when the favorite screen is shown favoriteButton

File: Youtube/Models/ReadFromRealm.swift [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+//  Youtube
+//
+//  Created by asiantech on 7/6/17.
+//  Copyright © 2017 AsianTech. All rights reserved.
+//
+
+import Foundation
+import RealmSwift
+
+class ReadFromRealm {
```
Comment (at-daiho) [inline Youtube/Models/ReadFromRealm.swift]: -> RealmManager

File: Youtube/Controllers/CategoryVC/DetailVC/DetailViewController.swift [hunk 6]
```diff
@@ -121,9 +143,7 @@ extension DetailViewController:UITableViewDelegate {
 
     func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
         let cellForHeader = tableView.dequeue(HeaderTableView.self)
```
Comment (at-daiho) [inline Youtube/Controllers/CategoryVC/DetailVC/DetailViewController.swift]: cellForHeader -> headerView

File: Youtube/Controllers/CategoryVC/DetailVC/DetailViewController.swift [hunk 7]
```diff
@@ -47,23 +44,48 @@ class DetailViewController: BaseViewController {
...
+            navigationItem.rightBarButtonItem?.tintColor = UIColor.white
+        }
+    }
+
+    // MARK: - delete UI favorite notification from screen favorite
+    @objc fileprivate func deleteFavorite(notification: NSNotification) {
+        navigationItem.rightBarButtonItem?.tintColor = UIColor.white
+        guard let videoNotification = notification.object as? Video else { return }
+        if video.id == videoNotification.id {
+            navigationItem.rightBarButtonItem?.tintColor = UIColor.white
```
Comment (at-daiho) [inline Youtube/Controllers/CategoryVC/DetailVC/DetailViewController.swift]: Please remove

File: Youtube/Controllers/CategoryVC/DetailVC/DetailViewController.swift [hunk 8]
```diff
@@ -24,21 +24,18 @@ class DetailViewController: BaseViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        // limit video display normal
```
Comment (at-daiho) [inline Youtube/Controllers/CategoryVC/DetailVC/DetailViewController.swift]: Show video in normal mode (not full screen)

File: Youtube/Controllers/CategoryVC/DetailVC/DetailViewController.swift [hunk 9]
```diff
@@ -24,21 +24,18 @@ class DetailViewController: BaseViewController {
...
+        videoPlayer.playerVars = ["playsinline": 1 as AnyObject]
     }
 
     override func configUI() {
         title = Strings.detailTitle
 
+        NotificationCenter.default.addObserver(self, selector: #selector(self.deleteFavorite(notification:)), name:NSNotification.Name("deleteFavoriteFromScreenFavorite"), object: nil)
```
Comment (at-daiho) [inline Youtube/Controllers/CategoryVC/DetailVC/DetailViewController.swift]: Please define

File: Youtube/Controllers/FavoriteVC/FavoriteViewController.swift [hunk 10]
```diff
@@ -16,32 +16,50 @@ class FavoriteViewController: BaseViewController {
...
+            RealmManager.shared.deleteAllVideo()
+            NotificationCenter.default.post(name: NSNotification.Name("deleteFavoriteFromScreenFavorite"), object: nil)
+            self.loadRealmData()
+            self.tableView.reloadData()
         }
     }
 
-    // TODO: - load realm
+    // MARK: - load realm
     fileprivate func loadRealmData() {
```
Comment (at-daiho) [inline Youtube/Controllers/FavoriteVC/FavoriteViewController.swift]: rename to getAllVideo()

File: Youtube/Controllers/SearchVC/SearchViewController.swift [hunk 11]
```diff
@@ -59,18 +60,24 @@ class SearchViewController: BaseViewController {
...
-    // TODO: - Dummy Data
-    fileprivate func createKeyword() {
-        for index in 1...8 {
-            let keyword = Keyword()
-            keyword.keyword = "Keyword \(index)"
-            keyword.lastTimeUse = Date()
-            keywords.append(keyword)
-        }
+    // MARK: - load realm
+    fileprivate func loadDataKeywords() {
```
Comment (at-daiho) [inline Youtube/Controllers/SearchVC/SearchViewController.swift]: Todo or delete

File: Youtube/Controllers/SearchVC/SearchViewController.swift [hunk 12]
```diff
@@ -79,14 +86,19 @@ class SearchViewController: BaseViewController {
...
+                searchAPI(keyword: text, isLoadMore: true)
+            }
         }
     }
 
     fileprivate func searchVideos(keyword: String) {
         searchBar?.text = keyword
         searchBar?.endEditing(true)
-        // TODO: - call api
+        let k = Keyword()
```
Comment (at-daiho) [inline Youtube/Controllers/SearchVC/SearchViewController.swift]: kw

File: Youtube/Define/Strings.swift [hunk 13]
```diff
@@ -20,5 +20,7 @@ class Strings {
     static let homeTitle = "Home"
     static let searchTitle = "Search"
     static let detailTitle = "Detail"
+    static let deleteTitle = "Delete all"
```
Comment (at-daiho) [inline Youtube/Define/Strings.swift]: deleteAll

File: Youtube/Define/Strings.swift [hunk 14]
```diff
@@ -20,5 +20,7 @@ class Strings {
...
     static let messageDelete = "Do you want delete all video?"
```
Comment (at-daiho) [inline Youtube/Define/Strings.swift]: confirmDelete

File: Youtube/Define/Strings.swift [hunk 15]
```diff
@@ -20,5 +20,7 @@ class Strings {
...
+    static let messageDeleteFavorite = "Do you want delete favorite?"
```
Comment (at-daiho) [inline Youtube/Define/Strings.swift]: cònirmDeleteFavorite

File: Youtube/Models/RealmManager.swift [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+            }
+        } catch {}
+    }
+
+    // MARK: - func delete video
+    func deleteVideo(_ video: Video) {
+        do {
+            let realm = try Realm()
+            let dbVideo = realm.object(ofType: Video.self, forPrimaryKey: video.id)
+            if dbVideo != nil {
```
Comment (at-daiho) [inline Youtube/Models/RealmManager.swift]: use if let

File: Youtube/Models/RealmManager.swift [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+            if dbKeyword == nil {
+                try realm.write {
+                    realm.add(keyword)
+                }
+            }
+        } catch {}
+    }
+
+    // MARK: - func read list keyword
+    func readKeywords(_ keywords:inout [Keyword]) {
```
Comment (at-daiho) [inline Youtube/Models/RealmManager.swift]: write func again

**Pull request merged**